resources/integration_test/testdata/dbmd_variable_result_set_data.yml

```diff
@@ -0,0 +1,20 @@
+dataset:
+  - db: integration_test
+    collection: foo
+    docs:
+      - {"_id": 0, "a": 1}
+      - {"_id": 1, "a": 2}
+    nonuniqueIndexes:
+      - {"a": 1}
```

Comment:
Added a way to define nonuniqueIndexes. Not sure if we want/need the ability to create general indexes with options. This is really just useful for one test so I'm ok with this special-purpose field. LMK if you feel differently!